src/main/groovy/edu/oregonstate/mist/cardsapi/resources/CardsResource.groovy

```diff
@@ -1,37 +1,49 @@
 package edu.oregonstate.mist.cardsapi.resources
 
 import io.dropwizard.jersey.params.IntParam
-import io.dropwizard.auth.Auth
 import edu.oregonstate.mist.cardsapi.core.Card
-import edu.oregonstate.mist.api.AuthenticatedUser
-
 import edu.oregonstate.mist.cardsapi.db.CardDAO
+import edu.oregonstate.mist.cardsapi.db.CardFluent
 import edu.oregonstate.mist.api.Resource
 import edu.oregonstate.mist.api.jsonapi.ResourceObject
 import edu.oregonstate.mist.api.jsonapi.ResultObject
-import org.slf4j.Logger
-import org.slf4j.LoggerFactory
 
+import javax.ws.rs.Consumes
+import javax.ws.rs.DELETE
+import javax.ws.rs.POST
 import javax.annotation.security.PermitAll
 import javax.ws.rs.GET
+import javax.ws.rs.PUT
 import javax.ws.rs.Path
 import javax.ws.rs.PathParam
 import javax.ws.rs.Produces
 import javax.ws.rs.core.Response
 import javax.ws.rs.core.MediaType
+import javax.ws.rs.QueryParam
+import javax.validation.Valid
+import com.google.common.base.Optional
+
+import org.skife.jdbi.v2.DBI
 
 // This will get a Card object from CardDAO and send responses for different endpoints
 
 @Path('/cards')
 @PermitAll
 @Produces(MediaType.APPLICATION_JSON)
 class CardsResource extends Resource {
-    Logger logger = LoggerFactory.getLogger(CardsResource.class)
 
     private final CardDAO cardDAO
+    private DBI dbi
+    CardFluent cardFluent = new CardFluent(dbi)
+
+    List<String> validTypes = ["skill", "attack", "power", "status", "curse"]
```

Comment:
What are your thoughts on moving these lists to DAO methods? By having these values hardcoded, you essentially have two sources of truth for types, colors, and rarities. This has its advantages since these values are stored in memory, checking against them is much faster than at the database. The disadvantage is that you're maintaining two sources of truth - the code and the DB - if you wanted to update or add a color, for example. What are your thoughts? I'm open to either way.